tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/localedistance/DistanceTable.java

```diff
@@ -0,0 +1,556 @@
+            }
+            return (key.getDistance() == node.distance);
+        }
+
+        /**
+         * Returns a sub-node corresponding to the given {@code (desired, supported)} mapping.
+         * If the node already exists, it is simply returned, otherwise a new node is created
+         * and any existing wildcard mappings are copied into it.
+         */
+        private Node getIntermediateNode(MappingKey key) {
+            Node node = getNode(key);
+            if (node == null) {
+                // This is expected to succeed because match rules are given in length
+                // order (i.e. language only before language+script etc.) and we always
+                // expect each group to end with an <ANY,ANY> mapping for the default
+                // distance. Thus, for any longer match rule, we should find (at least)
+                // the <ANY,ANY> node when looking for intermediate nodes.
+                //
+                // NOTE: Currently (desired==ANY) if-and-only-if (supported=ANY), so the
+                // only non-exact match we can get here is the <ANY,ANY> node. If we ever
+                // allow a mix of wildcard/non-wildcard keys, replace the getAnyNode() call
+                // with something like the line below:
+                // ----
+                // Node wildcardMatch = Iterables.find(
+                //      asList(getNode(desired, ANY), getNode(ANY, supported), getNode(ANY,ANY)),
+                //      Objects::nonNull);
+                // ----
+                Node wildcardMatch = getAnyNode();
+                checkState(wildcardMatch != null, "missing <ANY,ANY> mapping: %s", this);
+                // Default distances are the distance between any two *different* unknown
+                // subtags (so if the subtags are the same, the distance is zero).
+                int distance = key.getDesired().equals(key.getSupported()) ? 0 : wildcardMatch.distance;
+                node = new Node(distance);
+                node.copySubtablesFrom(wildcardMatch);
+                subtables.put(key.getDesired(), key.getSupported(), node);
+            }
+            return node;
+        }
+
+        /** Copies all subtable mappings from the given node into this one. */
+        private void copySubtablesFrom(Node src) {
+            checkState(subtables.isEmpty());
+            src.subtables.cellSet().forEach(
+                    c -> subtables.put(c.getRowKey(), c.getColumnKey(), new Node(c.getValue().distance)));
+        }
+
+        /**
+         * Writes all the mappings in the distance table sequentially to given Trie in sorted
+         * table order.
+         *
+         * <p>Mappings are written in a top-down recursive visitation with sub-tables inheriting
+         * the current prefix from parent tables via the given Trie span. At each level any
+         * mapped distances are written before recursing into the sub-tables.
+         */
+        private void writeTo(Span trie) {
```

Comment:
`Trie.Span trieSpan`

Reply:
Is that style preferred in ICU/CLDR?

Reply:
I don't know whether it's preferred, but for me, when a nested class name is very generic, I find the call sites confusing. So for me it's a judgment call, and for these I much prefer it as `Trie.Span`.

I would also like to ask you to add the "Span" suffixes to the variables/parameters, as suggested in my previous comments. It's not a "trie", it's a Span for a key-value pair that will eventually go into the trie. "trieSpan" seems like a good way to help the reader.

And I see you renamed ".with(str, fn)" to ".withSpan(str, fn)". Please revert that. With the suffix on the variable names, I think `Trie.Span trieSpan` and `trieSpan.with(str, fn)` are good.

Reply:
Ah, I misread your suggestion. Thought you wanted the method called "withDesiredSpan" which I shortened to "withSpan".